### PR TITLE
Added KubeContext to ClusterBuilders

### PR DESCRIPTION
**Pull request by AverageMarcus**
### What does this PR do?

- Adds a function to each ClusterBuilder that returns its supported KubeContext
- Adds a function to get a new ClusterBuilder instance based on a given KubeContext

### Checklist

- [x] CHANGELOG.md has been updated

### CODE REVIEW

**Review by mcharriere — COMMENTED** [review]
I'm a bit conflicted about this change. I understand that it might simplify usage from CI but it makes local usage more complicated.

that's it, if I understand correctly, because it needs now a crafted kubeconfig; so neither teleport nor opsctl can't be used directly without some kind of wrapper. Is this assumption correct?

**Comment by AverageMarcus** [comment]
> I'm a bit conflicted about this change. I understand that it might simplify usage from CI but it makes local usage more complicated.

That's always been the case. The documentation has always indicated that a specific format of kubeconfig was required.

File: kubeconfig
[the remark this answers is not in the document]
Comment (AverageMarcus) [reply kubeconfig:1]: Yeah, accidentally committed in my temp cluster files 🤦  Removed now.

File: results.json
[the remark this answers is not in the document]
Comment (AverageMarcus) [reply results.json:1]: Yeah, accidentally committed in my temp cluster files 🤦  Removed now.

**Comment by mcharriere** [comment]
> That's always been the case. The documentation has always indicated that a specific format of kubeconfig was required.

well, with the `standup` CLI; you were able to just set the E2E_CONTEXT env var to point to any context. I've been doing this:

```
PROVIDER="aws"
CONFIG="capa/standard"
CONTEXT="teleport.giantswarm.io-golem"

go run ${CODE_DIR}/cmd/standup/main.go \
      --provider ${PROVIDER} \
      --context ${CONTEXT} \
      --cluster-values ${CODE_DIR}/providers/${CONFIG}/test_data/cluster_values.yaml \
      --default-apps-values ${CODE_DIR}/providers/${CONFIG}/test_data/default-apps_values.yaml \
      --output ${OUTPUT_DIR}
```

and works fine. CODE_DIR is the cluster-test-suites repo path.

**Comment by AverageMarcus** [comment]
That's still supported. If you provide an unknown context it falls back to this code: https://github.com/giantswarm/cluster-standup-teardown/pull/10/files?diff=unified#diff-c2a7053ca5a17cef8e26a679050e3fb46d3f7efe3676057225a59bb5733093d1R114-R120

**Comment by AverageMarcus** [comment]
But, it's also worth pointing out that these CLIs are intended for use by the tests and we don't plan on supporting them beyond those use cases. `kubectl-gs` should be used for other purposes.

**Comment by mcharriere** [comment]
> That's still supported. If you provide an unknown context it falls back to this code

I see. Need to recheck that then.

**Pull request merged**